mock/src/tests.rs

```diff
@@ -28,6 +28,28 @@ macro_rules! assert_match {
     };
 }
 
+struct EventReceivers {
+    pub node_rx: Receiver<Event>,
+    pub client_rx: Receiver<Event>,
+}
+
+impl EventReceivers {
+    pub fn try_recv(&self) -> Result<Event, mpmc::TryRecvError> {
+        self.node_rx
+            .try_recv()
+            .or_else(|_| self.client_rx.try_recv())
```

Comment:
I'd also use `select` here, otherwise the operation is biased towards  `node_rx`.

Reply:
Would need to use try_select. does not seem really necessary for a test function.

Reply:
> Would need to use try_select

No, you would use `select!` with a `default` clause: 

```rust
select! {
    recv(self.node_rx) -> ev => ev,
    recv(self.client_rx) -> ev => ev,
    default => Err(TryRecvError::Empty)
}
```

Reply:
Agree that this is probably not necessary though.